it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,195 @@
+package com.google.cloud.teleport.it.datastream;
+
+import com.google.cloud.datastream.v1.*;
+import com.google.protobuf.FieldMask;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class DefaultDatastreamResourceManager implements DatastreamResourceManager {
+    private static final String AVRO_FILE_FORMAT = "avro";
+
+    private static final String JSON_FILE_FORMAT = "json";
+    private final DatastreamClient datastreamClient;
+    private final String location;
+    private final String projectId;
+    private final Map<String, ArrayList<String>> allowedTables;
+    private final String gcsBucketName;
+    private final String gcsRootPath;
+    private final boolean isOracleSource;
+    private final boolean isMySqlSource;
+    private final boolean isPostgresSource;
+    private final boolean isGcsDestination;
+    private final boolean isBqDestination;
+    private final String destinationOutputFormat = AVRO_FILE_FORMAT;
+
+    public DefaultDatastreamResourceManager(
+            DatastreamClient datastreamClient,
+            String location, String projectId,
+            Map<String, ArrayList<String>> allowedTables,
+            String gcsBucketName,
+            String gcsRootPath,
+            boolean isMySqlSource,
+            boolean isOracleSource,
+            boolean isPostgresSource,
+            boolean isBqDestination,
+            boolean isGcsDestination) {
```

Comment:
Strongly encourage using a builder to make the RM easier to use. Please look at other resource managers and @AutoValue annotations

Reply:
done. Please review